ExampleMod/Content/NPCs/ExampleCritter.cs

```diff
@@ -152,6 +152,21 @@ public override void HitEffect(NPC.HitInfo hit) {
 				Gore.NewGore(NPC.GetSource_Death(), NPC.position, NPC.velocity, Mod.Find<ModGore>($"{Name}_Gore_Head").Type, NPC.scale);
 				Gore.NewGore(NPC.GetSource_Death(), NPC.position, NPC.velocity, Mod.Find<ModGore>($"{Name}_Gore_Leg").Type, NPC.scale);
 			}
+
+		}
+
+		public override void OnHitByItem(Player player, Item item, NPC.HitInfo hit, int damageDone) {
+			ModContent.GetInstance<AdvancedExampleAchievement>().TotalDamageCondition.Value += damageDone;
+			if (item.type == ItemID.IronPickaxe) {
+				ModContent.GetInstance<AdvancedExampleAchievement>().IronPickaxeCondition.Complete();
+			}
+		}
+
+		public override void OnHitByProjectile(Projectile projectile, NPC.HitInfo hit, int damageDone) {
+			// OnHitByProjectile can be called on servers, so we need to check this to make sure to only access the achievement on a client to avoid null exceptions.
+			if (!Main.dedServ) {
```

Comment:
We generally prefer `Main.netMode != NetmodeID.Server`. Do we want an `owner` check to ignore traps and other player's projectiles?

Reply:
Made those changes. Docs say the method is only called on the owner anyway, so checking .owner is not needed.